sdk/loadtestservice/load-testing-rest/src/beginTestRunStatus.ts

```diff
@@ -0,0 +1,175 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { AbortController, AbortSignalLike } from "@azure/abort-controller";
+import { CancelOnProgress, OperationState, SimplePollerLike } from "@azure/core-lro";
+import { TestRunStatusPoller, PolledOperationOptions } from "./models";
+import { AzureLoadTestingClient } from "./index.js";
+import { TestRunGet200Response } from "./responses";
+import { isUnexpected } from "./isUnexpected";
+
+/**
+ * Creates a poller to poll for test run status.
+ * @param client - The Load Testing client.
+ * @param options - The operation options.
+ * @returns A poller which can be called to poll until completion of the job.
+ */
+export async function beginTestRunStatus(
+  client: AzureLoadTestingClient,
+  testRunId: string,
+  polledOperationOptions: PolledOperationOptions = {}
+): Promise<TestRunStatusPoller> {
+  let getTestRunResult = await client.path("/test-runs/{testRunId}", testRunId).get();
+  if (isUnexpected(getTestRunResult)) {
+    throw getTestRunResult.body.error;
+  }
+  //let TestRunStatusResponse: TestRunGet200Response;
+
+  type Handler = (state: OperationState<TestRunGet200Response>) => void;
+
+  const state: OperationState<TestRunGet200Response> = {
+    status: "notStarted",
+  };
+
+  const progressCallbacks = new Map<symbol, Handler>();
+  const processProgressCallbacks = async (): Promise<void> =>
+    progressCallbacks.forEach((h) => h(state));
+  let resultPromise: Promise<TestRunGet200Response> | undefined;
+  let cancelJob: (() => void) | undefined;
+  const abortController = new AbortController();
+  const currentPollIntervalInMs = polledOperationOptions.updateIntervalInMs ?? 2000;
+
+  const poller: SimplePollerLike<OperationState<TestRunGet200Response>, TestRunGet200Response> = {
+    async poll(_options?: { abortSignal?: AbortSignalLike }): Promise<void> {
+      getTestRunResult = await client.path("/test-runs/{testRunId}", testRunId).get();
+      if (isUnexpected(getTestRunResult)) {
+        throw getTestRunResult.body.error;
+      }
+
+      if (getTestRunResult.body.status === "FAILED") {
+        state.status = "failed";
+        state.error = new Error(getTestRunResult.body.status);
+      }
+
+      if (
+        getTestRunResult.body.status === "CANCELLING" ||
+        getTestRunResult.body.status === "CANCELLED"
+      ) {
+        state.status === "canceled";
+      }
+
+      if (getTestRunResult.body.status === "DONE") {
+        state.status = "succeeded";
+        state.result = getTestRunResult;
+      }
+
+      await processProgressCallbacks();
+
+      if (state.status === "canceled") {
+        throw new Error("Operation was canceled");
+      }
+      if (state.status === "failed") {
+        throw state.error;
+      }
```

Comment:
In a rest-level client, we don't want to throw exceptions. The customer is expected instead to inspect the `status` field themselves to determine whether the operation has failed/canceled.
```suggestion
```

Reply:
updated.